conf/config.py

```
@@ -0,0 +1,136 @@
+            errors.append(f"Invalid NSE_DOWNLOAD_TIMEOUT: {self.NSE_DOWNLOAD_TIMEOUT}")
+        
+        if self.NSE_RETRY_ATTEMPTS < 0:
+            errors.append(f"Invalid NSE_RETRY_ATTEMPTS: {self.NSE_RETRY_ATTEMPTS}")
+        
+        # Validate log level
+        valid_log_levels = ['DEBUG', 'INFO', 'WARNING', 'ERROR', 'CRITICAL']
+        if self.LOG_LEVEL not in valid_log_levels:
+            errors.append(f"Invalid LOG_LEVEL: {self.LOG_LEVEL}. Must be one of {valid_log_levels}")
+        
+        if errors:
+            raise ValueError(f"Configuration validation failed:\n" + "\n".join(errors))
```

Ruff (0.14.10)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)